FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW/arc/arc_exception.h

```diff
@@ -0,0 +1,293 @@
+/* ------------------------------------------
+ * Copyright (c) 2022, Synopsys, Inc. All rights reserved.
+
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+
+ * 1) Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+
+ * 2) Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+
+ * 3) Neither the name of the Synopsys, Inc., nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+   --------------------------------------------- */
+
+/**
+ * @file
+ * @ingroup ARC_HAL_EXCEPTION_CPU ARC_HAL_EXCEPTION_INTERRUPT
+ * @brief header file of exception and interrupt management module
+ */
+
+#ifndef _ARC_HAL_EXCEPTION_H_
+#define _ARC_HAL_EXCEPTION_H_
+
+#include "arc.h"
+#include "arc_builtin.h"
+
+#ifndef __ASSEMBLY__
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup ARC_HAL_EXCEPTION_CPU
+ * @{
+ */
+#ifndef NUM_EXC_CPU
+/*!< number of CPU exceptions */
+#define NUM_EXC_CPU             16
+#endif
+
+/*!< total number of exceptions */
+#define NUM_EXC_ALL             (NUM_EXC_CPU + NUM_EXC_INT)
+
+typedef struct int_exc_frame {
+	uint32_t erbta;
+
+	uint32_t r30;   /* r30 is useless, skipped? */
+	uint32_t ilink; /* r29 is useless, skipped?*/
+	/* r28 is sp, saved other place */
+	uint32_t fp;    /* r27 */
+	uint32_t gp;    /* r26 */
+
+	uint32_t r12;
+	uint32_t r0, r1, r2, r3;
+	uint32_t r4, r5, r6, r7, r8, r9;
+	uint32_t r10, r11;
+	uint32_t blink; /* r31 */
+	uint32_t lp_end, lp_start, lp_count;
+	uint32_t ret;
+	uint32_t status32;
+} EMBARC_PACKED INT_EXC_FRAME_T;
```

Comment:
Let's simply use `__attribute__((packed))` here. Same in other places for EMBARC_PACKED.